pkg/operator/options/suite_test.go

```diff
@@ -36,17 +39,47 @@ func TestOptions(t *testing.T) {
 }
 
 var _ = Describe("Options", func() {
-	Context("Parse", func() {
-		It("should correctly parse feature gates", func() {
-			opts, err := options.New().Parse("--feature-gates", "Drift=true")
-			Expect(err).To(BeNil())
-			Expect(opts.FeatureGates.Drift).To(BeTrue())
-			opts, err = options.New().Parse("--feature-gates", "Drift=false")
-			Expect(err).To(BeNil())
-			Expect(opts.FeatureGates.Drift).To(BeFalse())
+	BeforeEach(func() {
+		fs = flag.NewFlagSet("karpenter", flag.ContinueOnError)
+		opts = &options.Options{}
+		opts.AddFlags(fs)
+	})
+
+	Context("FeatureGates", func() {
+		DescribeTable(
+			"should successfully parse well formed feature gate strings",
+			func(gateStr string, driftVal bool) {
+				gates, err := options.ParseFeatureGates(gateStr)
+				Expect(err).To(BeNil())
+				Expect(gates.Drift).To(Equal(driftVal))
+			},
+			Entry("Basic true", "Drift=true", true),
+			Entry("Basic false", "Drift=false", false),
+			Entry("With whitespace", "Driftt\t= false", false),
+			Entry("With additional gates", "Hello=true,Drift=false,World=true", false),
+		)
+		It("should successfully parse well formed feature gate str", func() {
+			for _, entry := range []struct {
+				gateStr  string
+				driftVal bool
+			}{
+				{"Drift = true", true},
```

Comment:
`DescribeTable`?

Reply:
It was originally to take advantage of BeforeEach but I don't think it's necessary after the interface changes. It probably does make sense to switch to a for loop.

Reply:
I pretty strongly disagree against this syntax. This is the syntax of `go test` tests but is not the recommended way to create tests that have structure like this. If you find yourself creating a struct to store test parameters/expected results, that's a great use-case for `DescribeTable`.

Reply:
See: https://onsi.github.io/ginkgo/#table-specs

Reply:
Also, the other good reason to do the `DescribeTable` methodology is that when one of the test cases fails, you will have a description built-in to the failure vs. no detail with what you have right now

Reply:
So are you saying that should be a `DescribeTable`? That's what it was when you left the original comment, I changed it after the fact. I'm happy to change it back to a `DescribeTable`, whatever works best stylistically.